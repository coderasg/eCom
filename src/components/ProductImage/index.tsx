import React from 'react';

import styles from './productImage.module.scss';

type Props = {
  src: string;
  bannerText?: string;
  name?: string;
  className?: string;
};

function ProductImage(props: Props) {
  const { src , name, bannerText, className} = props;
  
  return (
    <div className={`${styles.productImag} ${className}`}>
      {bannerText && <div className={styles.banner}>
        <span className={styles.bannerText}>{bannerText}</span>
      </div>}
      <img data-testid="ProductImage-wrapper" src={src} className={styles.productImage} />
      {name && <div className={styles.title}>{`${name}`}</div>}
    </div>
  );
}

export default ProductImage;
