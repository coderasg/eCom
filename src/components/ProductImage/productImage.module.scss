@use "./../../variables";


.productImag {
  position: relative;

  .title {
    padding: variables.$paddingMid;;
    font-size: 25px;
  }
  
  .banner {
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 0px 62px 62px 62px;
    border-color: transparent transparent variables.$highlightColor transparent;
    transform: rotate(313deg);
    position: absolute;
    top: -9px;
    left: -42px;
    right: 0;

    .bannerText {
      position: absolute;
      top: 28px;
      left: -20px;
      color: variables.$white;
    }
  }
}
