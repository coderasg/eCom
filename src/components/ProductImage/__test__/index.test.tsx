import React from 'react';
import renderer from 'react-test-renderer';

import ProductImage from '../index';

describe('(Component) ProductImage', () => {
  const defaultProps = {
    src: 'src',
  };
  
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<ProductImage {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
