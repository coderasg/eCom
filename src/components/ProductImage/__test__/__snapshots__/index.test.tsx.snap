// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductImage should pass snapshot test 1`] = `
<div
  className="undefined undefined"
>
  <img
    data-testid="ProductImage-wrapper"
    src="src"
  />
</div>
`;
